src/main/java/com/example/demo/services/AuctionService.java

```diff
@@ -0,0 +1,79 @@
+package com.example.demo.services;
+
+import com.example.demo.dtos.AuctionDTO;
+import com.example.demo.entities.Auction;
+import com.example.demo.entities.User;
+import com.example.demo.products.Product;
+import com.example.demo.repositories.AuctionRepository;
+import com.example.demo.repositories.UserRepository;
+import jakarta.mail.MessagingException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.util.Streamable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static com.example.demo.services.NotificationService.notifyAuctionClosed;
+
+@Service
+@RequiredArgsConstructor
+public class AuctionService {
+    private final AuctionRepository auctionRepository;
+    private final UserRepository userRepository;
+
+    @Transactional
+    public void createAuction(User owner, AuctionDTO dto, Float startPrice, String title) {
+        List<Product> products = dto.getProducts();
+        owner.getProducts().removeAll(products);
+        userRepository.save(owner);
+        Auction auction = new Auction();
+        auction.setClosed(false);
+        auction.setOwner(owner);
```

Comment:
should owner and auction be saved in cascade?

Reply:
Are there any possible alternatives where it is guaranteed that we won't lose owner products?